src/parser/jobs/blu/modules/RevengeBlast.tsx

```diff
@@ -0,0 +1,221 @@
+		this.revengeBlastWindows.closeCurrent(this.parser.pull.timestamp + this.parser.pull.duration)
+
+		// Revenge Blasts that happened above the HP threshold.
+		this.suggestions.add(new TieredSuggestion({
+			icon: this.data.actions.REVENGE_BLAST.icon,
+			content: <Trans id="blu.revenge_blast.bad.content">
+				Only use <DataLink action="REVENGE_BLAST" /> if your HP will be below 20% by the time the slidecast window starts.
+			</Trans>,
+			tiers: REVENGE_SEVERITY.CAST_OUT_OF_WINDOW,
+			value: this.badRevengeBlasts,
+			why: <Trans id="blu.revenge_blast.bad.why">
+				<Plural value={this.badRevengeBlasts ?? 0} one="# Revenge Blast cast" other="# Revenge Blast casts" /> happened when above 20% HP
+			</Trans>,
+		}))
+
+		// If you know a Revenge Blast window is coming, using Whistle first is a DPS gain.
+		const unwhistledWindows: number = this.revengeBlastWindows.entries
+			.filter(revengeWindow => revengeWindow.data.hadRevengeBlast && !revengeWindow.data.hadWhistle)
+			.length
+		this.suggestions.add(new TieredSuggestion({
+			icon: this.data.actions.REVENGE_BLAST.icon,
+			content: <Trans id="blu.revenge_blast.unwhistled.content">
+				If it won't drop a <DataLink action="REVENGE_BLAST" /> cast, you should cast <DataLink action="WHISTLE" /> before entering a <DataLink action="REVENGE_BLAST" showIcon={false} /> window for a DPS gain over using your filler.
+			</Trans>,
+			tiers: REVENGE_SEVERITY.UNWHISTLED_OPENING_REVENGE_BLAST,
+			value: unwhistledWindows,
+			why: <Trans id="blu.revenge_blast.unwhistled.why">
+				<Plural value={unwhistledWindows ?? 0} one="# Revenge Blast window" other="# Revenge Blast windows" /> did not have the <DataLink action="WHISTLE" /> buff.
+			</Trans>,
+		}))
+	}
+
+	override output() {
+		if (this.revengeBlastWindows.entries.length === 0) { return undefined }
+
+		const gcdHeader = {
+			header: <Trans id="blu.revenge_blast.rotation-table.header.blast-count">Revenge Blasts</Trans>,
+			accessor: 'gcds',
+		}
+
+		const gcd = this.gcd.getDuration() / 1000
+		const revengeBlastID = this.data.actions.REVENGE_BLAST.id
+		const dat = this.data
+		const downtime = this.downtime
+		const rotationData: RotationTableEntry[] = []
+		this.revengeBlastWindows.entries
+			.filter(revengeWindow => ((revengeWindow.end ?? revengeWindow.start) - revengeWindow.start) > REVENGE_BLAST_MINIMUM_WINDOW_MS)
+			.forEach(revengeWindow => {
+				const revengeStart = revengeWindow.start - this.parser.pull.timestamp - REVENGE_BLAST_CAST_TIME + SLIDECAST_OFFSET
+				const revengeEnd   = (revengeWindow.end ?? revengeWindow.start) - this.parser.pull.timestamp - REVENGE_BLAST_CAST_TIME + SLIDECAST_OFFSET
+				const forcedDowntime = downtime.getDowntime(
+					revengeWindow.start,
+					revengeWindow.end ?? revengeWindow.start,
+				)
+				// TODO: this is a cop-out.  Dealing with downtime is hard.
+				if (forcedDowntime > REVENGE_BLAST_MINIMUM_WINDOW_MS) { return }
+
+				const deltaMs      = revengeEnd - revengeStart - forcedDowntime
```

Comment:
I'm mildly surprised the linter isn't complaining about this, but adding all the extra whitespace between the `const varName` and the `= <assignment>` chunks makes this harder to read. Can we get rid of that (and double-check the other files as well, since I saw the same pattern elsewhere)?

Reply:
Oh that's my work linter for another language firing by mistake; it was trying to whitespace-align operators, ala
```
const foo     = 1
const barquux = 2
```

Anyway, should be removed as of 58930091f8629afd9498f63502065dffba1a8507